pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -269,7 +270,18 @@ func (rc *reconciler) attachDesiredVolumes() {
 			nodes := rc.actualStateOfWorld.GetNodesForVolume(volumeToAttach.VolumeName)
 			if len(nodes) > 0 {
 				if !volumeToAttach.MultiAttachErrorReported {
-					simpleMsg, detailedMsg := volumeToAttach.GenerateMsg("Multi-Attach error", "Volume is already exclusively attached to one node and can't be attached to another")
+					podNames := rc.desiredStateOfWorld.GetVolumePodsOnNodes(nodes, volumeToAttach.VolumeName)
+					var simpleMsg, detailedMsg string
```

Comment:
Should we exclude current pod that is trying to attach this volume? Perhaps it does not matter. Also, it may be worth not "leaking" information about pods that do not exist in current user's namespace.

Reply:
Added a separate commit with elaborate logic that does not reveal pod names to users in another namespace and on the other way logs all pods to controller-manager logs.